### DISCUSSION TITLE
Limit concurrent connections/requests

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2723] [Ideas]
Is there any way to limit **Concurrent** connections/requests to storage?

Some providers limit these in order to provide good pricing per GiB, and any requests beyond this (say, 20) connections/requests done simultaneously (`GET, HEAD, LIST, PUT...`) will return HTTP 429 or 503 or similar return codes. An option like `--max-requests 20` could be a great adition in order to tune JuiceFS according Storage specifications and stay below provider limits (whatever they are).

**Comment by xiaogaozi** [comment]
The `--max-uploads` and `--max-deletes` options can limit the concurrency of uploads and deletes. But there is currently no option to limit the concurrency of all requests.

**Reply by solracsf** [reply]
Ok thanks. Is there any place in the code where this can be hardcoded instead of passing it as an option?

**Reply by xiaogaozi** [reply]
Refer to here: https://github.com/juicedata/juicefs/blob/dbf1fe1ed731284cff43029930bcd56363faffa8/cmd/mount.go#L321-L322
